src/Autofac/Core/Resolving/ResolveOperationBase.cs

```diff
@@ -246,32 +184,102 @@ public object GetOrCreateInstance(ISharingLifetimeScope currentOperationScope, R
             return requestContext.Instance!;
         }
 
+        /// <summary>
+        /// Enter a new dependency chain block where subsequent requests inside the operation are allowed to repeat
+        /// registrations from before the block.
+        /// </summary>
+        /// <returns>A disposable that should be disposed to exit the block.</returns>
+        public IDisposable EnterNewDependencyDetectionBlock() => RequestStack.EnterSegment();
+
+        /// <summary>
+        /// Invoke this method to execute the operation for a given request.
+        /// </summary>
+        /// <param name="request">The resolve request.</param>
+        /// <returns>The resolved instance.</returns>
+        protected object ExecuteOperation(ResolveRequest request)
+        {
+            object result;
+
+            try
+            {
+                InitiatingRequest = request;
+                if (DiagnosticSource.IsEnabled())
+                {
+                    DiagnosticSource.OperationStart(this, request);
+                }
+
+                result = GetOrCreateInstance(CurrentScope, request);
+            }
+            catch (ObjectDisposedException disposeException)
+            {
+                if (DiagnosticSource.IsEnabled())
+                {
+                    DiagnosticSource.OperationFailure(this, disposeException);
+                }
+
+                throw;
+            }
+            catch (DependencyResolutionException dependencyResolutionException)
+            {
+                if (DiagnosticSource.IsEnabled())
+                {
+                    DiagnosticSource.OperationFailure(this, dependencyResolutionException);
+                }
+
+                End(dependencyResolutionException);
+                throw;
+            }
+            catch (Exception exception)
+            {
+                End(exception);
+                if (DiagnosticSource.IsEnabled())
+                {
+                    DiagnosticSource.OperationFailure(this, exception);
+                }
+
+                throw new DependencyResolutionException(ResolveOperationResources.ExceptionDuringResolve, exception);
+            }
+            finally
+            {
+                ResetSuccessfulRequests();
+            }
+
+            End();
+
+            if (DiagnosticSource.IsEnabled())
+            {
+                DiagnosticSource.OperationSuccess(this, result);
+            }
+
+            return result;
+        }
+
         /// <summary>
         /// Basic pipeline invocation steps used when retrieving an instance. Isolated
         /// to enable it to be optionally surrounded with diagnostics.
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private void InvokePipeline(ResolveRequest request, ResolveRequestContext requestContext)
+        private void InvokePipeline(ResolveRequest request, ResolveRequestContext requestContextContext)
```

Comment:
Sorry, I missed this naming issue first time out...

Reply:
Search & Replace does not always work as desired :roll_eyes: